### PR TITLE
Gps settings:  Allow user control of autoconfig

### LINKED ISSUE #1394: Specific config options for Ublox 8 series
The Ublox 8 modules allow for simultaneous use of GPS and GLONASS, also a higher rate with a measurement period of 56ms.
The following two messages can be incorporated into the ublox autoconfig to take advantage of these possibilities.

`CFG GNSS`
`0000|  B5 62 06 3E 2C 00 00 00 20 05 00`
`000B|  08 10 00 01 00 01 01 01 01 03 00`
`0016|  01 00 01 01 03 08 10 00 00 00 01`
`0021|  01 05 00 03 00 01 00 01 01 06 08`
`002C|  0E 00 01 00 01 01 FF 3D`   

`CFG RATE`
`0000|  B5 62 06 08 06 00 38 00 01 00 01`
`000B|  00 4E 0A`

### PR DESCRIPTION
**Pull request by mlyle**
Allows the user to enable simultaneous GPS+GLONASS and thus resolves #1394.  Also corrects version detection logic, enables GCS to see greater numbers of satellites.

Most possible combinations of settings verified to work on hardware.

Future work will be in the GCS, making the GPS widget prettier when more satellites are present, and adding options to save the configuration to the GPS.

### CODE REVIEW

**Comment by mlyle** [comment]
Note I am not able to test the SBAS settings for except None and WAAS for correct behavior.  (I am in North America).

File: ground/gcs/src/plugins/gpsdisplay/gpsconstellationwidget.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ GpsConstellationWidget::GpsConstellationWidget(QWidget *parent) : QGraphicsView(
     setScene(scene);
 
     // Now create 'maxSatellites' satellite icons which we will move around on the map:
-    for (int i=0; i < MAX_SATTELITES;i++) {
+    for (int i=0; i < MAX_SATELLITES;i++) {
```
Comment (guilhermito) [inline ground/gcs/src/plugins/gpsdisplay/gpsconstellationwidget.cpp]: spaces around "=" and after last ";" while you are at it? :)

File: ground/gcs/src/plugins/gpsdisplay/gpssnrwidget.cpp [hunk 2]
```diff
@@ -7,7 +7,7 @@ GpsSnrWidget::GpsSnrWidget(QWidget *parent) :
     setScene(scene);
 
     // Now create 'maxSatellites' satellite icons which we will move around on the map:
-    for (int i=0; i < MAX_SATTELITES;i++) {
+    for (int i=0; i < MAX_SATELLITES;i++) {
```
Comment (guilhermito) [inline ground/gcs/src/plugins/gpsdisplay/gpssnrwidget.cpp]: and here

File: shared/uavobjectdefinition/gpssatellites.xml [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-        <field name="SatsInView" units="" type="int8" elements="1"/>
-        <field name="PRN" units="" type="int8" elements="16"/>
-        <field name="Elevation" units="degrees" type="float" elements="16"/>
-        <field name="Azimuth" units="degrees" type="float" elements="16"/>
-        <field name="SNR" units="" type="int8" elements="16"/>
+        <field name="SatsInView" units="" type="uint8" elements="1"/>
+        <field name="PRN" units="" type="uint8" elements="30"/>
+        <field name="Elevation" units="degrees" type="int8" elements="30"/>
+        <field name="Azimuth" units="degrees" type="int16" elements="30"/>
+        <field name="SNR" units="" type="int8" elements="30"/>
```
Comment (guilhermito) [inline shared/uavobjectdefinition/gpssatellites.xml]: dB?

**Comment by guilhermito** [comment]
Looks good!

**Comment by mlyle** [comment]
Changes applied, thanks!

**Comment by mlyle** [comment]
Note one negative thing with this change is increasing the maximum size of the satellites vector by quite a bit, which spends buffer space.

**Comment by mlyle** [comment]
Resolves issue #1394

**Comment by CheBuzz** [comment]
Tested on Quanton with NEO M8N and on Sparky1 with LEA 5H.  Both worked great.  I think this is good to merge.

**Pull request merged**